plugins/services/src/js/reducers/serviceForm/FormReducers/Artifacts.js

```diff
@@ -0,0 +1,44 @@
+import {
+  ADD_ITEM,
+  REMOVE_ITEM,
+  SET
+} from '../../../../../../../src/js/constants/TransactionTypes';
+
+function processTransaction(state, {type, path, value}) {
```

Comment:
You mean duplication of the code in the JSON reducer? Well I had an interessting discussion about that the other day. Basically I was also thinking this is duplication, but it actually isn't. With one function we are creating the json output and with the other one we are creating the Form output. In this case these both happen to be the same. So if one changes the other one will still work.